api/files/views.py

```diff
@@ -35,7 +35,249 @@ def get_file(self, check_permissions=True):
+        name          string             name of the file or folder; use for display
+        kind          string             "file" or "folder"
+        path          url path           unique path for this entity, used in "move" actions
+        size          integer            size of file in bytes, null for folders
+        provider      string             storage provider for this file. "osfstorage" if stored on the OSF.  Other
+                                         examples include "s3" for Amazon S3, "googledrive" for Google Drive, "box"
+                                         for Box.com.
+        last_touched  iso8601 timestamp  last time the metadata for the file was retrieved. only applies to non-OSF
+                                         storage providers.
+
+    ##Relationships
+
+    ###Checkout (*files, folders*)
+
+    ** *Checkouts are still a work-in-progress. This property may be set but is currently a no-op.* **
+
+    A link to the user who has checked out the file.  While a file is checked out, only the checkout holder may modify
+    it.  If a file is not checked out, this url will be null.
+
+    ###Files (*folders*)
+
+    The `files` endpoint lists all of the subfiles and folders of the current folder. Will be null for files.
+
+    ###Versions (*files*)
+
+    The `versions` endpoint provides version history for files.  Will be null for folders.
+
+    ##Links
+
+        info:        the canonical api endpoint for the latest version of the file
+        new_folder:  url to target when creating new subfolders (null for files)
+        move:        url to target for move, copy, and rename actions
+        upload:      url to target for uploading new files and updating existing files
+        download:    url to request a download of the latest version of the file (null for folders)
+        delete:      url to target for deleting files and folders
+
+    ##Actions
+
+    The `links` property of the response provides endpoints for common file operations. The currently-supported actions
+    are:
+
+    ###Get Info (*files, folders*)
+
+        Method:   GET
+        URL:      links.info
+        Params:   <none>
+        Success:  200 OK + file representation
+
+    The details of a particular file can be retrieved by performing a GET request against the `info` link. The response
+    will be a standard OSF response format with the [OSF File attributes](#attributes).
+
+    ###Download (*files*)
+
+        Method:   GET
+        URL:      links.download
+        Params:   <none>
+        Success:  200 OK + file body
+
+    To download a file, issue a GET request against the `download` link.  The response will have the Content-Disposition
+    header set, which will will trigger a download in a browser.
+
+    ###Create Subfolder (*folders*)
+
+        Method:       PUT
+        URL:          links.new_folder
+        Query Params: ?name={new_folder_name}
```

Comment:
`?kind=folder&name={new_folder_name}`
Note: `kind=folder` is already included in the url